### DISCUSSION TITLE
What is the alternative of "findMultiple" ?

### DISCUSSION THREAD

**Discussion by Jawaher-Cherif** [stephanrauh/ngx-extended-pdf-viewer#1888] [Q&A]
Hello, 
I am trying to update the version of `ngx-extended-pdf-viewer` from `15.2.0` to `18.0.1` and I am having trouble replacing the function `findMultiple` in my code because it doesn’t exist anymore in `18.0.1` version.
What is the alternative of `findMultiple` ? how to search in pdf using an array of strings ?

**Comment by stephanrauh** [comment]
I'm afraid I removed it. There's no replacement. To be honest, that was a decision I didn't do easily. I hated it. The problem was that the find algorithm of pdf.js evolved rapidly, and I didn't manage to merge my changes into their advanced algorithm. After trying a year or so, I gave up. I pruned the library in order to keep up with the progress of the base library.

Technically speaking, it's possible to implement the feature again. All you need to do is to analyse the `FindController` class of pdf.js and to add `findMultiple` to the algorithm again. At the moment, I prefer to wait until the dust has settled, and the rapid evolvement of the `FindController` has slowed down again.

**Reply by Jawaher-Cherif** [reply]
Thanks a lot for your response.

I switched to version `17.4.6` so I can use the function `findMultiple`, but unfortunately the pdf is no longer loaded.
And I have this error in my console: 
`  Uncaught SyntaxError: Cannot use import statement outside a module   viewer-3.5.550.min.js:1  `

Is this version not compatible with Angular 16 ?

**Reply by stephanrauh** [reply]
Damn. That rings a bell. Did you run into this error? https://github.com/stephanrauh/ngx-extended-pdf-viewer/issues/1864

If so, I can probably publish a bugfix. Just tell me if it's the same error.

**Reply by Jawaher-Cherif** [reply]
Well I am having the error in the `17.4.7` version, but I think that's the same error 
https://github.com/stephanrauh/ngx-extended-pdf-viewer/blob/17.4.7/projects/ngx-extended-pdf-viewer/assets/viewer-3.5.550.min.js

This is the file that cause the error , is it possible to remove the `import` in this file ( in this version ) please ?

Thank you.

**Reply by stephanrauh** [reply]
Consider it done. Just grant me a little delay. Currently, I'm busy preparing the slide for [this talk](https://javascript-conference.com/node-js/graaljs-javascript-engine/). I don't want to let you down, but I don't believe I can spend much time on my pet project before the end of the month.

**Reply by Jawaher-Cherif** [reply]
Hello, 
Is there any update on this issue ?

**Reply by stephanrauh** [reply]
I've removed the import statement in the current version.

**Reply by Jawaher-Cherif** [reply]
Hello,
Thank you, is it possible to remove the import in the `17.4.7` version ? because I can't use the current version in my project ?

**Reply by lixaotec** [reply]
any updates on this?

**Reply by stephanrauh** [reply]
I've removed the import two months ago. If you update to a current 17.4.* version, it should be gone.
<img width="970" alt="image" src="https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/3045767/df846c2d-f0e1-4778-a021-038642c36694">

**Reply by lixaotec** [reply]
dear @stephanrauh , i´ve updated to 18+ because im using vite (esbuild) with angular 16. 
Unfortunately ngx extended pdf v17 got issues with cannot import viewer outside a module.. than I had to decide go up in version to v18, lefting findmultiple feature.. ( i also cannot update angular to v17 yet)
I suppose this is a no way to have both... isnt it?
appreciate your help. tks!

**Reply by stephanrauh** [reply]
Of course there's a way to have both... I'm just afraid you won't like it. Actually, there are two options. First, you can provide a pull request for adding findMultiple etc. to version 18. That's not trivial, but I can give you some guidance, so I guess that's an option. Alternatively, you can talk my boss into doing this stuff at work. However, that involves paying my compony money, and it's not cheap. On the plus side, every customer who paid me for doing open-source at work in the past was happy. They still are!

**Reply by Jawaher-Cherif** [reply]
dear @stephanrauh , I tested the version `17.4.9` but I still have the error of the import 
<img width="852" alt="Screenshot 2023-11-28 at 11 38 26" src="https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/135248924/a31770b9-512f-462a-9fc9-e7bc43c4070d">
Which version should I install ?

**Reply by stephanrauh** [reply]
@Jawaher-Cherif Please update to version 17.4.10. To my surprise, there were two import statements, and I only remove the first one.
